src/components/Menu/MenuItem/__stories__/MenuItem.stories.js

```diff
@@ -72,6 +71,28 @@ export const Label = {
   }
 };
 
+export const SubMenu = {
+  render: () => (
+    <Menu>
+      <MenuItem title="Opens on item hover" onClick={() => alert("clicked on menu item")}>
```

Comment:
Currently click on the `MenuItem` instead of triggering the alert open-close the menu, which feels weird
For the split case it works as expected

Reply:
![ezgif-3-7a4fbecfa7](https://github.com/mondaycom/monday-ui-react-core/assets/104433616/2323b22f-7181-4773-9cf1-0b4b060f411f)